### PR TITLE
Fix secret/configmap/projected volume update tests to work for large clusters

### PR DESCRIPTION
**Pull request by shyamjvs**
Fixes https://github.com/kubernetes/kubernetes/issues/48359

/cc @kubernetes/sig-node-pr-reviews @wojtek-t @gmarek

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -2421,7 +2421,29 @@ func WaitForAllNodesSchedulable(c clientset.Interface, timeout time.Duration) er
...
+	secretTTL, found := GetNodeTTLAnnotationValue(c)
+	if !found {
+		Logf("Couldn't get node ttl annotation")
+	}
+	podLogTimeout := 240*time.Second + secretTTL
+	return podLogTimeout
+}
+
+func GetNodeTTLAnnotationValue(c clientset.Interface) (time.Duration, bool) {
+	nodes, err := c.Core().Nodes().List(metav1.ListOptions{})
```
Comment (wojtek-t) [inline test/e2e/framework/util.go]: Please don't silently drop errors. Instead:
```
ExpectNoError(err)
if len(nodes.Items) == 0 {
  ExpectNoError(...)
}
```

File: test/e2e/common/configmap.go [hunk 2]
```diff
@@ -73,12 +72,8 @@ var _ = framework.KubeDescribe("ConfigMap", func() {
...
 
 	It("updates should be reflected in volume [Conformance] [Volume]", func() {
-
-		// We may have to wait or a full sync period to elapse before the
-		// Kubelet projects the update into the volume and the container picks
-		// it up. This timeout is based on the default Kubelet sync period (1
-		// minute) plus additional time for fudge factor.
-		const podLogTimeout = 300 * time.Second
+		podLogTimeout := framework.GetPodSecretUpdateTimeout(f.ClientSet)
+		mt_retry_time := fmt.Sprintf("--retry_time=%v", int(podLogTimeout.Seconds()))
```
Comment (gmarek) [inline test/e2e/common/configmap.go]: This is very non-go name - please use camel case. Plus I don't know what 'mt' stands for.

Comment (shyamjvs) [inline test/e2e/common/configmap.go]: Done. Old habit from C++ times ;)

File: test/e2e/framework/util.go (hunk 1 above)
Comment (shyamjvs) [inline test/e2e/framework/util.go]: Done.

**Comment by shyamjvs** [comment]
The e2e tests are failing due to not being able to find out TTL annotation on node. I'll look into it.

**Comment by shyamjvs** [comment]
The test is failing because of absence of TTL annotation on the node:

```
I0704 03:38:39.803] Jul  4 10:19:54.828: INFO: Annotations on node map[volumes.kubernetes.io/controller-managed-attach-detach:true]
```

The reason is that being node-e2e test we don't seem to start controller-manager and hence the ttl-controller couldn't set the annotation.
However previously we were just defaulting to '0' in case of no annotation, but now we're failing due to ExpectNoError.
@wojtek-t Are you fine with defaulting to 0 and just logging the error?

**Comment by wojtek-t** [comment]
> @wojtek-t Are you fine with defaulting to 0 and just logging the error?

SGTM

**Comment by shyamjvs** [comment]
@Random-Liu Maybe the "node.alpha.kubernetes.io/ttl" annotation should be set manually on the nodes in the node-e2e tests.

**Comment by shyamjvs** [comment]
@wojtek-t Fixed it. PTAL

File: test/e2e/framework/util.go [hunk 3]
```diff
@@ -2421,20 +2421,38 @@ func WaitForAllNodesSchedulable(c clientset.Interface, timeout time.Duration) er
...
 
-func GetTTLAnnotationFromNode(node *v1.Node) (time.Duration, bool) {
+func GetPodSecretUpdateTimeout(c clientset.Interface) time.Duration {
+	// With SecretManager(ConfigMapManager), we may have to wait up to full sync period +
+	// TTL of secret(configmap) to elapse before the Kubelet projects the update into the
+	// volume and the container picks it up.
+	// So this timeout is based on default Kubelet sync period (1 minute) + maximum TTL for
+	// secret(configmap) that's based on cluster size + additional time as a fudge factor.
+	secretTTL, err := GetNodeTTLAnnotationValue(c)
+	Logf("Couldn't get node TTL annotation: %v", err)
```
Comment (wojtek-t) [inline test/e2e/framework/util.go]: Log it only when err != nil

Comment (shyamjvs) [inline test/e2e/framework/util.go]: Oops.. SOrry for the blooper. Fixed it.

File: test/e2e/framework/util.go [hunk 4]
```diff
@@ -2421,20 +2421,40 @@ func WaitForAllNodesSchedulable(c clientset.Interface, timeout time.Duration) er
...
-func GetTTLAnnotationFromNode(node *v1.Node) (time.Duration, bool) {
+func GetPodSecretUpdateTimeout(c clientset.Interface) time.Duration {
+	// With SecretManager(ConfigMapManager), we may have to wait up to full sync period +
+	// TTL of secret(configmap) to elapse before the Kubelet projects the update into the
+	// volume and the container picks it up.
+	// So this timeout is based on default Kubelet sync period (1 minute) + maximum TTL for
+	// secret(configmap) that's based on cluster size + additional time as a fudge factor.
+	secretTTL, err := GetNodeTTLAnnotationValue(c)
+	if err != nil {
+		Logf("Couldn't get node TTL annotation: %v", err)
```
Comment (wojtek-t) [inline test/e2e/framework/util.go]: Sorry for being boring, but can you also please explicitly log that we will be using "0" as a default value here?

Comment (shyamjvs) [inline test/e2e/framework/util.go]: Done.

**Pull request merged**